src/main/java/ca/corefacility/bioinformatics/irida/ria/web/projects/metadata/ProjectSampleMetadataAjaxController.java

```diff
@@ -0,0 +1,257 @@
+package ca.corefacility.bioinformatics.irida.ria.web.projects.metadata;
+
+import java.io.*;
+import java.util.*;
+import java.util.Map.Entry;
+
+import javax.servlet.http.HttpSession;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+
+import ca.corefacility.bioinformatics.irida.exceptions.EntityNotFoundException;
+import ca.corefacility.bioinformatics.irida.exceptions.MetadataImportFileTypeNotSupportedError;
+import ca.corefacility.bioinformatics.irida.model.project.Project;
+import ca.corefacility.bioinformatics.irida.model.sample.MetadataTemplateField;
+import ca.corefacility.bioinformatics.irida.model.sample.Sample;
+import ca.corefacility.bioinformatics.irida.model.sample.metadata.MetadataEntry;
+import ca.corefacility.bioinformatics.irida.ria.utilities.SampleMetadataStorage;
+import ca.corefacility.bioinformatics.irida.ria.web.projects.ProjectControllerUtils;
+import ca.corefacility.bioinformatics.irida.ria.web.services.UIMetadataImportService;
+import ca.corefacility.bioinformatics.irida.service.ProjectService;
+import ca.corefacility.bioinformatics.irida.service.sample.MetadataTemplateService;
+import ca.corefacility.bioinformatics.irida.service.sample.SampleService;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Files;
+
+/**
+ * This class is designed to be used for bulk actions on {@link MetadataEntry}
+ * within a {@link Project}.
+ */
+@Controller
+@RequestMapping("/ajax/projects/sample-metadata")
+public class ProjectSampleMetadataAjaxController {
```

Comment:
Another thing we have been doing is creating UI services which contain the actual code. From the controllers we just make the call to the UI service to do the work. This way we keep the controllers lean. There's a bunch of UI service files we've created that you can use as a guide. These UI service files are located in `src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services`

Reply:
Please see 14b368cc1ca1cb55444c451aa229eef92dba31d9.